networkx/convert_matrix.py

```diff
@@ -1170,83 +1171,38 @@ def to_numpy_array(
 
     if nodelist is None:
         nodelist = list(G)
-        nodeset = G
-        nlen = len(G)
-    else:
-        nlen = len(nodelist)
-        nodeset = set(G.nbunch_iter(nodelist))
-        if nlen != len(nodeset):
-            for n in nodelist:
-                if n not in G:
-                    raise nx.NetworkXError(f"Node {n} in nodelist is not in G")
-            raise nx.NetworkXError("nodelist contains duplicates.")
-
-    undirected = not G.is_directed()
-    index = dict(zip(nodelist, range(nlen)))
+    nlen = len(nodelist)
+
+    # Input validation
+    nodeset = set(nodelist)
+    if nodeset - set(G):
+        raise nx.NetworkXError(f"Node {nodeset - set(G)} in nodelist is not in G")
```

Comment:
```suggestion
        raise nx.NetworkXError(f"Nodes {nodeset - set(G)} in nodelist is not in G")
```
Change of grammar with multiple nodes reported.